src/screens/RequestSwap/SwapScreen.tsx

```diff
@@ -1,109 +1,199 @@
-    const [selectedIndex, setSelectedIndex] = useState(new IndexPath(0));
+    const [selectedPeopleIndices, setSelectedPeopleIndices] = useState<IndexPath[]>([]);
+    const [selectedTimeIndices, setSelectedTimeIndices] = useState<IndexPath[]>([]);
+    const [selectedDates, setSelectedDates] = useState([]);
 
-    //this is the date they chose to swap on
-    const selectedSwapDate = useSelector((state) => state.swapReducer.date);
-    const dispatch = useDispatch();
+    //https://stackoverflow.com/questions/48834275/good-way-to-chain-filter-functions-in-javascript/48834470#48834470
+    const filters = [
+        (task) =>
+            selectedPeopleIndices.length === 0
+                ? true
+                : selectedPeopleIndices.some(
+                      (item) =>
+                          swapCandidates[item.row].firstName === task.user.firstName &&
+                          swapCandidates[item.row].lastName === task.user.lastName,
+                  ),
+        (task) =>
+            selectedDates.length > 0
+                ? selectedDates.some((date) =>
+                      compareDates(new Date(date), new Date(task.date)),
+                  )
+                : true,
+        (task) => !task.isSelected,
+    ];
 
-    //true if radio is checked or (if an index and a date are both selected)
-    let nextButtonClickable =
-        checked || (selectedIndex !== null && selectedSwapDate !== null);
+    //these operations only need to be run once on render. useEffect runs after everything is loaded so things that relied on tasks got angry at me
+    const tasks = React.useMemo(
+        () =>
+            swapCandidates
+                .reduce((acc, currentValue) => [...acc, ...currentValue.tasks], [])
+                .map((task) => ({ ...task, isSelected: false }))
+                .sort((a, b) => new Date(a.date).getTime() - new Date(b.date).getTime()), //may not need if swapCandidates are pre-sorted
+        [swapCandidates],
+    );
 
-    const onNextHandler = () => {
-        if (checked) {
-            dispatch(selectTargetTask(undefined));
-        } else {
-            dispatch(selectTargetTask(displayValue.tasks[0]));
-        }
-        props.navigation.navigate('SwapSummary');
+    const [filteredTasks, setFilteredTasks] = useState(tasks);
+    const selectedTasks = filteredTasks.filter((task) => task.isSelected);
+
+    //callbacks
+    const onItemSelect = (item, index) => {
+        setFilteredTasks((f) => {
+            let tempFilteredTasks = [...f];
+            tempFilteredTasks[index].isSelected = !tempFilteredTasks[index].isSelected;
+            return tempFilteredTasks;
+        });
+    };
+    const onTilePress = (date) => {
+        const dates = selectedDates.filter((item) => !compareDates(item, date));
+        dates.length === selectedDates.length
+            ? setSelectedDates([...selectedDates, date])
+            : setSelectedDates(dates);
```

Comment:
I'm checking to see if a newly selected date has already been selected.

After filtering the dates array by a new date, if the length of the filtered array is the same, that means it isn't already in that array, so i want to add it to the array.

If the length is different, then i want to return the filtered array because that date was previously in there, but now it has been removed

Reply:
Ah that makes sense.